src/components/IFrame.js

```diff
@@ -17,73 +28,155 @@ function getNewDotURL(url) {
+        return `${params.viewMode === 'charts' ? ROUTES.INSIGHTS_OLDDOT : ROUTES.EXPENSES_OLDDOT}${paramString ? `/?param=${paramString}` : ''}`;
     }
 
     if (pathname === 'admin_policies') {
         const {section} = params;
-        return section === 'individual' ? 'individual_workspaces' : 'group_workspaces';
+        return section === 'individual' ? ROUTES.INDIVIDUALS_OLDDOT : ROUTES.GROUPS_OLDDOT;
     }
 
     if (pathname === 'policy') {
-        const workspaceID = params.policyID || '';
-        const section = urlObj.hash.slice(1) || 'overview';
+        const workspaceID = params.policyID ?? '';
+        const section = urlObj.hash.slice(1) ?? '';
+
+        let sectionName = section;
+        switch (section) {
+            case '':
+                sectionName = 'overview';
+                break;
+            case 'js_policyEditor_perDiem':
+                sectionName = 'per-diem';
+                break;
+            case 'exportFormats':
+                sectionName = 'export-formats';
+                break;
+            default:
+                sectionName = section;
+        }
 
-        return `workspace/${workspaceID}/${section}`;
+        return ROUTES.WORKSPACES_OLDDOT.getRoute(workspaceID, sectionName);
     }
 
     if (pathname === 'settings') {
         const {section} = params;
-        return `settings/${section}`;
+
+        let sectionName = '';
+        switch (section) {
+            case undefined:
+            case '':
+                sectionName = 'account';
+                break;
+            case 'expenserules':
+                sectionName = 'expense-rules';
+                break;
+            case 'creditcards':
+                sectionName = 'cards';
+                break;
+            default:
+                sectionName = section;
+        }
+
+        return `settings/${sectionName}`;
     }
 
-    if (pathname.includes('domain')) {
-        return pathname;
+    if (pathname === 'admin_domains') {
+        return ROUTES.DOMAINS_OLDDOT;
     }
 
-    return pathname;
-}
+    if (pathname.includes('domain')) {
+        const [, section] = pathname.split('_');
 
-function getOldDotURL(url) {
-    const urlObj = new URL(url);
-    const pathname = urlObj.pathname;
-    const paths = pathname.slice(1).split('/');
+        let sectionName = '';
+        switch (section) {
+            case '':
+            case 'companycards':
+                sectionName = 'cards';
+                break;
+            case 'loadingDock':
+                sectionName = 'tools';
+                break;
+            default:
+                sectionName = section;
+        }
 
-    // TODO: temporary measure until linking config is adjusted
-    if (pathname.startsWith('/r')) {
-        return 'inbox';
+        return ROUTES.DOMAIN_OLDDOT.getRoute(sectionName);
     }
 
-    if (pathname === 'home') {
+    return pathname;
+}
+
+function getOldDotURL(screenName, params) {
```

Comment:
We need to update this function to always append the params `isInNewDot=true&isDarkMode=true`